IdentityCore/src/workplacejoin/MSIDPkeyAuthHelper.m

```diff
@@ -140,8 +144,27 @@ + (NSString *)createDeviceAuthResponse:(NSString *)audience
         return nil;
     }
     NSArray *arrayOfStrings = @[[NSString stringWithFormat:@"%@", [[identity certificateData] base64EncodedStringWithOptions:0]]];
+    NSError *error;
+    NSString *alg = [[MSIDKeyOperationUtil sharedInstance] getJwtAlgorithmForKey:identity.privateKeyRef context:nil error:&error];
+
+    if ([NSString msidIsStringNilOrBlank:alg])
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Can't determine key signing alg for private key : %@", error);
```

Comment:
Assuming defaulting to RS256, we can do that but it will be helpful only when we alg=RS256 and there is a bug in detecting alg as RS256. For existing RSA device key I didn't see an issue/bug detecting the alg so we should be okay here